README.md
# simple_list_tile

A fully customisable Flutter simple list tile with gradients.

<div style="display:flex">
<img width="355" alt="alert2" src="https://user-images.githubusercontent.com/44444254/145979456-6bb152cc-1906-420f-b2a8-b0ec1615b1a6.png" width="200">
<div/>

<b>Usage without gradient: </b>

            SimpleListTile(
              onTap: () {
                print('test');
              },
              title: Text(
                'Title',
                style: TextStyle(
                  color: Colors.white,
                  fontWeight: FontWeight.bold,
                  fontSize: 20,
                ),
              ),
              subtitle: Text(
                'Subtitle',
                style: TextStyle(
                  color: Colors.white,
                ),
              ),
              trailing: Icon(
                Icons.arrow_forward_ios,
                color: Colors.white,
              ),
              leading: Icon(
                Icons.phone_android,
                color: Colors.blue,
              ),
              borderRadius: BorderRadius.circular(20),
              tileColor: Colors.grey[300]!,
              circleColor: Colors.grey[100]!,
              circleDiameter: 200,
              gradient: LinearGradient(
                colors: [Colors.red, Colors.yellow],
              ),
            ),
            SizedBox(
              height: 20,
            ),
            SimpleListTile(
              onTap: () {
                print('test');
              },
              title: Text(
                'Title',
                style: TextStyle(
                  color: Colors.black,
                  fontWeight: FontWeight.bold,
                  fontSize: 20,
                ),
              ),
              subtitle: Text(
                'Subtitle',
                style: TextStyle(
                  color: Colors.black,
                ),
              ),
              trailing: Icon(Icons.arrow_forward_ios),
              leading: Image.asset(
                'assets/guided_faq.png',
              ),
              borderRadius: BorderRadius.circular(20),
              tileColor: Colors.white,
              circleColor: Colors.blue[800]!,
              circleDiameter: 200,
            ),


<b>Usage with gradient : </b>

                SimpleListTile(
                onTap: () {
                print('test');
              },
              title: Text(
                'Title',
                style: TextStyle(
                  color: Colors.white,
                  fontWeight: FontWeight.bold,
                  fontSize: 20,
                ),
              ),
              subtitle: Text(
                'Subtitle',
                style: TextStyle(
                  color: Colors.white,
                ),
              ),
              trailing: Icon(
                Icons.arrow_forward_ios,
                color: Colors.white,
              ),
              leading: Icon(
                Icons.phone_android,
                color: Colors.blue,
              ),
              borderRadius: BorderRadius.circular(20),
              tileColor: Colors.grey[300]!,
              circleColor: Colors.grey[100]!,
              circleDiameter: 200,
              gradient: LinearGradient(
                colors: [Colors.blue, Colors.green],
              ),
            ),


CHANGELOG.md
- 0.0.3: example updated
- 0.0.2: onTap added
- 0.0.1: gradient added

lib/simple_list_tile.dart
library simple_list_tile;

import 'package:flutter/material.dart';

class SimpleListTile extends StatelessWidget {
  final Widget title;
  final Widget? subtitle;
  final Widget? trailing;
  final Widget leading;
  final Color tileColor;
  final Color circleColor;
  final EdgeInsets padding;
  final BorderRadius borderRadius;
  final double circleDiameter;
  final Gradient? gradient;
  final VoidCallback? onTap;

  SimpleListTile({
    required this.title,
    this.subtitle,
    this.trailing,
    this.tileColor = Colors.grey,
    this.circleColor = Colors.white,
    this.padding = const EdgeInsets.all(8),
    required this.borderRadius,
    required this.leading,
    this.circleDiameter = 80,
    this.gradient,
    required this.onTap,
  });
  @override
  Widget build(BuildContext context) {
    final double _circleDiameter = circleDiameter > 100 ? 80 : circleDiameter;

    return SafeArea(
      child: GestureDetector(
        onTap: onTap,
        child: Container(
          // color: Colors.blue,
          padding: EdgeInsets.only(right: 0),
          child: LayoutBuilder(builder: (context, constraints) {
            return Stack(
              clipBehavior: Clip.none,
              alignment: Alignment.center,
              children: [
                Container(
                  height: _circleDiameter + 10,
                  margin: EdgeInsets.only(left: 40),
                  padding: padding,
                  decoration: BoxDecoration(
                    color: tileColor,
                    borderRadius: borderRadius,
                    gradient: gradient,
                  ),
                  child: Padding(
                    padding: EdgeInsets.only(left: _circleDiameter - 40),
                    child: Row(
                      crossAxisAlignment: CrossAxisAlignment.center,
                      children: [
                        Expanded(
                          child: Column(
                            mainAxisAlignment: MainAxisAlignment.center,
                            crossAxisAlignment: CrossAxisAlignment.start,
                            children: [
                              title,
                              SizedBox(
                                height: 5.0,
                              ),
                              subtitle ?? SizedBox.shrink(),
                            ],
                          ),
                        ),
                        trailing ?? SizedBox.shrink()
                      ],
                    ),
                  ),
                ),
                Positioned(
                  left: -constraints.maxWidth + _circleDiameter,
                  right: 0,
                  child: Container(
                    height: _circleDiameter,
                    width: _circleDiameter,
                    clipBehavior: Clip.antiAlias,
                    decoration: BoxDecoration(
                      color: circleColor,
                      shape: BoxShape.circle,
                    ),
                    child: leading,
                  ),
                )
              ],
            );
          }),
        ),
      ),
    );
    return SafeArea(
      child: Container(
        // color: Colors.blue,
        padding: EdgeInsets.only(right: 0),
        child: LayoutBuilder(builder: (context, constraints) {
          return Stack(
            clipBehavior: Clip.none,
            alignment: Alignment.center,
            children: [
              Container(
                height: _circleDiameter + 10,
                margin: EdgeInsets.only(left: 40),
                padding: padding,
                decoration: BoxDecoration(
                  color: tileColor,
                  borderRadius: borderRadius,
                  gradient: gradient,
                ),
                child: Padding(
                  padding: EdgeInsets.only(left: _circleDiameter - 40),
                  child: Row(
                    crossAxisAlignment: CrossAxisAlignment.center,
                    children: [
                      Expanded(
                        child: Column(
                          mainAxisAlignment: MainAxisAlignment.center,
                          crossAxisAlignment: CrossAxisAlignment.start,
                          children: [
                            title,
                            SizedBox(
                              height: 5.0,
                            ),
                            subtitle ?? SizedBox.shrink(),
                          ],
                        ),
                      ),
                      trailing ?? SizedBox.shrink()
                    ],
                  ),
                ),
              ),
              Positioned(
                left: -constraints.maxWidth + _circleDiameter,
                right: 0,
                child: Container(
                  height: _circleDiameter,
                  width: _circleDiameter,
                  clipBehavior: Clip.antiAlias,
                  decoration: BoxDecoration(
                    color: circleColor,
                    shape: BoxShape.circle,
                  ),
                  child: leading,
                ),
              )
            ],
          );
        }),
      ),
    );
  }
}


example/lib/screens/home_screen.dart
import 'package:flutter/material.dart';
import 'package:simple_list_tile/simple_list_tile.dart';

class HomeScreen extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: Colors.grey[100],
      appBar: AppBar(),
      body: Padding(
        padding: const EdgeInsets.all(8.0),
        child: Column(
          children: [
            SimpleListTile(
              onTap: () {
                print('test');
              },
              title: Text(
                'Title',
                style: TextStyle(
                  color: Colors.white,
                  fontWeight: FontWeight.bold,
                  fontSize: 20,
                ),
              ),
              subtitle: Text(
                'Subtitle',
                style: TextStyle(
                  color: Colors.white,
                ),
              ),
              trailing: Icon(
                Icons.arrow_forward_ios,
                color: Colors.white,
              ),
              leading: Icon(
                Icons.phone_android,
                color: Colors.blue,
              ),
              borderRadius: BorderRadius.circular(20),
              tileColor: Colors.grey[300]!,
              circleColor: Colors.grey[100]!,
              circleDiameter: 200,
              gradient: LinearGradient(
                colors: [Colors.blue, Colors.green],
              ),
            ),
            SizedBox(
              height: 20,
            ),
            SimpleListTile(
              onTap: () {
                print('test');
              },
              title: Text(
                'Title',
                style: TextStyle(
                  color: Colors.white,
                  fontWeight: FontWeight.bold,
                  fontSize: 20,
                ),
              ),
              subtitle: Text(
                'Subtitle',
                style: TextStyle(
                  color: Colors.white,
                ),
              ),
              trailing: Icon(
                Icons.arrow_forward_ios,
                color: Colors.white,
              ),
              leading: Icon(
                Icons.phone_android,
                color: Colors.blue,
              ),
              borderRadius: BorderRadius.circular(20),
              tileColor: Colors.grey[300]!,
              circleColor: Colors.grey[100]!,
              circleDiameter: 200,
              gradient: LinearGradient(
                colors: [Colors.red, Colors.yellow],
              ),
            ),
            SizedBox(
              height: 20,
            ),
            SimpleListTile(
              onTap: () {
                print('test');
              },
              title: Text(
                'Title',
                style: TextStyle(
                  color: Colors.black,
                  fontWeight: FontWeight.bold,
                  fontSize: 20,
                ),
              ),
              subtitle: Text(
                'Subtitle',
                style: TextStyle(
                  color: Colors.black,
                ),
              ),
              trailing: Icon(Icons.arrow_forward_ios),
              leading: Image.asset(
                'assets/guided_faq.png',
              ),
              borderRadius: BorderRadius.circular(20),
              tileColor: Colors.white,
              circleColor: Colors.blue[800]!,
              circleDiameter: 200,
            ),
          ],
        ),
      ),
    );
  }
}


example/lib/main.dart
import 'package:flutter/material.dart';
import 'package:simple_list_tile/simple_list_tile.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      debugShowCheckedModeBanner: false,
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: Scaffold(
        backgroundColor: Colors.grey[100],
        appBar: AppBar(),
        body: Padding(
          padding: const EdgeInsets.all(8.0),
          child: Column(
            children: [
              SimpleListTile(
                onTap: () {
                  print('test');
                },
                title: Text(
                  'Title',
                  style: TextStyle(
                    color: Colors.white,
                    fontWeight: FontWeight.bold,
                    fontSize: 20,
                  ),
                ),
                subtitle: Text(
                  'Subtitle',
                  style: TextStyle(
                    color: Colors.white,
                  ),
                ),
                trailing: Icon(
                  Icons.arrow_forward_ios,
                  color: Colors.white,
                ),
                leading: Icon(
                  Icons.phone_android,
                  color: Colors.blue,
                ),
                borderRadius: BorderRadius.circular(20),
                tileColor: Colors.grey[300]!,
                circleColor: Colors.grey[100]!,
                circleDiameter: 200,
                gradient: LinearGradient(
                  colors: [Colors.blue, Colors.green],
                ),
              ),
              SizedBox(
                height: 20,
              ),
              SimpleListTile(
                onTap: () {
                  print('test');
                },
                title: Text(
                  'Title',
                  style: TextStyle(
                    color: Colors.white,
                    fontWeight: FontWeight.bold,
                    fontSize: 20,
                  ),
                ),
                subtitle: Text(
                  'Subtitle',
                  style: TextStyle(
                    color: Colors.white,
                  ),
                ),
                trailing: Icon(
                  Icons.arrow_forward_ios,
                  color: Colors.white,
                ),
                leading: Icon(
                  Icons.phone_android,
                  color: Colors.blue,
                ),
                borderRadius: BorderRadius.circular(20),
                tileColor: Colors.grey[300]!,
                circleColor: Colors.grey[100]!,
                circleDiameter: 200,
                gradient: LinearGradient(
                  colors: [Colors.red, Colors.yellow],
                ),
              ),
              SizedBox(
                height: 20,
              ),
              SimpleListTile(
                onTap: () {
                  print('test');
                },
                title: Text(
                  'Title',
                  style: TextStyle(
                    color: Colors.black,
                    fontWeight: FontWeight.bold,
                    fontSize: 20,
                  ),
                ),
                subtitle: Text(
                  'Subtitle',
                  style: TextStyle(
                    color: Colors.black,
                  ),
                ),
                trailing: Icon(Icons.arrow_forward_ios),
                leading: Image.asset(
                  'assets/guided_faq.png',
                ),
                borderRadius: BorderRadius.circular(20),
                tileColor: Colors.white,
                circleColor: Colors.blue[800]!,
                circleDiameter: 200,
              ),
            ],
          ),
        ),
      ),
    );
  }
}
